c-johnson/skystrife-public
# [CANV-XXX] Enc chat

![image](https://github.com/c-johnson/skystrife-public/assets/583272/5c5b3979-d6b6-4f23-8617-4b42ff70558e)

Things to do still:
  - [x] Encrypt messages
    - [x] swap `eth-sig-utils` for https://github.com/paulmillr/noble-ciphers
  - [x] Switch between messages types in the lobby
  - [x] Update player name 
  - [x] Cleanup tasks:
      - [x] Use a client-specific library for message encrypting/decrypting, remove node-polyfills()
      - [x] Switch out css code for tailwind classes
  - [x] Test and review

To test:
  - Check out the "Season 2" working commit (018fe4c8310bae00)
  - Boot up a working game server per the [README](https://github.com/latticexyz/skystrife-public/blob/main/README.md). 
    - Two changes I needed locally, ymmv; 
      - 1) I removed `run-pty` from devDependencies and installed it manually 
      - 2) Flip `auto-install-peers=true` in the .npmrc 
  - Log in as the root user, mint yourself a season pass, and create a new 2 player game.
  - In a separate browser, log in as a dummy user by adding `?asPlayer` to the query params. 
  - Both players should boot up a game session, ready up, and be launched into the game. Both chat windows, "Player" and "All", should appear. 

Remaining issues: 
  - None
